example/strcpy.s

```diff
@@ -6,14 +6,14 @@ strcpy:
       mv a2, a0             # Copy dst
       li t0, -1             # Infinite AVL
 loop:
-    vsetvli x0, t0, e8, m8, ta,ma  # Max length vectors of bytes
+    vsetvli x0, t0, e8, m8, ta, ma # Max length vectors of bytes
     vle8ff.v v8, (a1)        # Get src bytes
       csrr t1, vl           # Get number of bytes fetched
     vmseq.vi v1, v8, 0      # Flag zero bytes
     vfirst.m a3, v1         # Zero found?
       add a1, a1, t1        # Bump pointer
-    vmsif.m v0, v1          # Set mask up to and including zero byte.
-    vse8.v v8, (a2), v0.t    # Write out bytes
+    vsetvli x0, a3, e8, m8, ta, ma # Setup vl including zero byte.
+    vse8.v v8, (a2)          # Write out bytes
```

Comment:
I'm skeptical that this is an improvement in general.  Deferring the dependence on `a3` until the end of the loop allows the `vse` to start executing before the `vfirst` result is returned.

For decoupled implementations, there is a loss of decoupling in either case, but deferring the loss-of-decoupling event to the end of the loop improves ILP.

Reply:
@aswaterman 
The original version is better in benchmark.
The stall time for "vfirst->vsetvli" is bigger than the original implementation.